examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,297 @@
+"""
+Title: Document Images Denoiser.
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/01/18
+Last modified: 2022/01/27
+Description: Example of Document Image Denoiser coupled with Tesseract OCR engine to improve text readability.
+"""
+"""
+## Introduction
+
+This example explains a "Image Denoiser" using UNet architecture. ```Autoencoders``` are
+mostly employed for ```Image Restoration``` problems but in recent times ```U-Net```
+based architectures with ```skip-connections``` have gained popularity for several
+Image-to-Image tasks. we aim to solve the problem with simple ```Pretrained U-Net as
+Encoder``` and ```Efficient Sub-Pixel CNN as Decoder```. The problem we focus here is
+enhancing the Document Images which are deteriorated through external degradations like
+blur, corrupt text blocks etc... ; we have followed up on few seminal papers which are
+presented below for reference. At the end of this tutorial user will gain clear
+understanding of building custom Image Pre-Processors using deep learning that helps us
+to mitigate real world OCR issues and extending it for variouse Image-to-Image
+applications.
+
```

Comment:
@anish9 thanks for the update. This is much better than before but still needs a few touches. 

- Starting with a quote like - **This example explains an "Image Denoiser" using UNet architecture....** - It may make this example a potential repeat of the following existing [code example](https://keras.io/examples/vision/autoencoder/).  But if I understand it's not.  In an OCR system, often the algorithm suffers due to the noise effect on text. And that reduces the performance of the OCR system. So, a possible solution would be to add a text image preprocessing method for the OCR system. You can look into [this](https://stackoverflow.com/q/64808986/9215780) for some references. In general, I think it's ideal to start with formulating the problem, current research progress followed by what will be shown here, etc.

- It's not needed to use code-syntax highlighter like  ```my_text```. To highlight any text if needed, we can just bold it with **my_text**. Also, the sentence starts with, **The problem we focus ..... for reference** - too long to read. Make a full stop at here [**.. blocks etc...; we ...**].